### PR TITLE
remove logout code (bug 1119428)

### PR DESCRIPTION
**Pull request by washort**

### CODE REVIEW

File: public/js/views/reset-start.js [hunk 1]
```diff
@@ -68,27 +68,7 @@ define([
...
-      logger.log('starting logout timer.');
-      this.resetLogoutTimeout = window.setTimeout(function() {
-        // If the log-out times-out then abort/reject the requests/deferred.
-        logger.log('logout timed-out');
-        authResetUser.abort();
-        personaLogout.reject();
-      }, settings.logout_timeout);
-
-      $.when(authResetUser, personaLogout)
+      $.when(auth.resetUser())
```
Comment (muffinresearch) [inline public/js/views/reset-start.js]: If there's just one thing here you don't need the when,

File: public/js/views/reset-start.js [hunk 2]
```diff
@@ -68,27 +68,7 @@ define([
...
-      } else {
-        personaLogout = this.logoutPersona();
-      }
-
-      if (this.resetLogoutTimeout) {
-        window.clearTimeout(this.resetLogoutTimeout);
-      }
-
-      logger.log('starting logout timer.');
-      this.resetLogoutTimeout = window.setTimeout(function() {
```
Comment (muffinresearch) [inline public/js/views/reset-start.js]: I think it would make sense to leave in the timeout for our own logout in-case that fails.

**Comment by muffinresearch** [comment]
Hmm tests didn't run - we'll want to make sure they're green before landing this.

**Comment by muffinresearch** [comment]
If we don't need the logoutPersona func can we remove it in this branch? Also this seems like a good opportunity to clean up func names like handlePersonaLogin given it's not Persona any more.

File: public/js/views/reset-start.js [hunk 3]
```diff
@@ -68,27 +68,7 @@ define([
...
-        logger.log('logout timed-out');
-        authResetUser.abort();
-        personaLogout.reject();
-      }, settings.logout_timeout);
-
-      $.when(authResetUser, personaLogout)
+      $.when(auth.resetUser())
         .done(function _allLoggedOut() {
           logger.log('Clearing logout reset timer.');
           window.clearTimeout(that.resetLogoutTimeout);
```
Comment (muffinresearch) [inline public/js/views/reset-start.js]: This is related to the timeout code removed above.

File: tests/static/testlib/stubbyid.js [hunk 4]
```diff
@@ -206,6 +206,11 @@
     },
     request: function navigator_id_request(options) {
       options = options || {};
+      // treating any refreshAuthentication value as a demand for a password
+      // prompt
+      if (options.refreshAuthentication !== undefined) {
+        setLoginState(null, false);
```
Comment (muffinresearch) [inline tests/static/testlib/stubbyid.js]: Comment is handy - would be worth adding some logging here too.

File: public/js/views/reset-start.js (hunk 2 above)
Comment (washort) [inline public/js/views/reset-start.js]: good point, since authReset sticks around, this should too

File: tests/static/testlib/stubbyid.js [hunk 5]
```diff
@@ -206,6 +206,13 @@
     },
     request: function navigator_id_request(options) {
       options = options || {};
+      // treating any refreshAuthentication value as a demand for a password
+      // prompt
+      if (options.refreshAuthentication !== undefined) {
+        log("Setting login state to null since refreshAuthentication " +
+            "was requested");
+        setLoginState(null, false);
```
Comment (muffinresearch) [inline tests/static/testlib/stubbyid.js]: Is there anything else in this file that needs to be updated here to better match NativeFxA? How much state management needs to stay?

**Comment by muffinresearch** [comment]
Looks good aside from the need to clean-up leftover personaisms. (that can come later if it's just about naming)

If you're happy this is working and the web flow is still good - r+wc

**Comment by washort** [comment]
I'll file a separate bug for the state management and naming.

**Comment by muffinresearch** [comment]
> I'll file a separate bug for the state management and naming.

Perfect.

**Pull request merged**